lumicks/pylake/nb_widgets/kymotracker_widgets.py

```diff
@@ -482,6 +459,76 @@ def show(self, use_widgets, **kwargs):
         self._select_state({"value": "mode", "old": "", "new": "Track lines"})
 
 
+def _get_node_info(lines):
+    """Build an array of information on all nodes in all tracked lines.
+
+    Parameters
+    ----------
+    lines: KymoLineGroup
+        Tracked lines.
+
+    Returns
+    -------
+    np.ndarray
+        Array of shape [n, 4] with each row representing information about
+        a node in a tracked line corresponding to:
+        (track index, node index within track, x-coordinate, y-coordinate)
+    """
+
+    nodes = [
+        np.vstack(
+            (
+                np.full(len(track), j),
+                np.arange(len(track)),
+                track.seconds,
+                track.position,
+            )
+        )
+        for j, track in enumerate(lines)
+    ]
+    return np.hstack(nodes).T
+
+
+def _get_nearest(lines, nodes, x, y, scale, cutoff_radius=0.05):
```

Comment:
Same here. Would be nice to have a test for this. There's also ample room for optimization here, and it'd be nice to have a test in place if we wanna have a go at optimizing this.